### ISSUE TITLE
[google_sign_in] Inconsistent login behavior in development and production

### ISSUE THREAD

**Issue by kkondarev** [flutter/flutter#101433]
Hello, I am using the google_sign_in plugin to handle user login. Everything works fine in the emulator and also on locally build .apk installed on device. Exactly same code/configuration, but build as .abb and installed via Google Play cannot log in (I am uploading now a new version to gather more information, like possible exceptions).

However, any idea what could provoke such a behavior?

Thank you
Kras

**Comment by darshankawar** [comment]
@kkondarev 
Would be good to get an error log or exception log that will help to debug further.

**Comment by kkondarev** [comment]
Yes, waiting for the next app update to be approved in Play Store... takes a bit...

Locally everything works flawlessly, this is what bugs me - both in emulator and as locally installed apk the login works without any issue, only when coming from Play Store - not.
Was asking if anyone has experienced such strange behavior and can point me in some direction.

**Comment by darshankawar** [comment]
@kkondarev 
Maybe you can try to run `flutter run --release` and then try to login and see if it throws any log ?

Check if this for reference : https://github.com/firebase/flutterfire/issues/7435

**Comment by kkondarev** [comment]
Thanks for the reply. Nope, this works - building a release app (.apk) and testing it (also on a phone) works perfectly fine.

Meanwhile the app was approved on Play Store (lucky coincidence it happened today!), so I was able to find out, that during login the following exception was raised: PlatformException(sign_in_failed, com.google.android.gms.common.api.b: 10: , null, null)
I double-checked my SHA-1, scopes, etc - all good. Long story short, after a lot of googleing I found out, that the issue is caused by the Play Store's app signing. Replaced the SHA-1 in the API console with the one generated by Play Store and now everything works awesomely again.
